sensor/common/virtualmachine/component.go

```diff
@@ -0,0 +1,27 @@
+package virtualmachine
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/sensor/common"
+)
+
+// Component provides functionality to send virtual machines to Central.
+type Component interface {
+	common.SensorComponent
+
+	Send(ctx context.Context, vm *storage.VirtualMachine) error
+}
+
+// NewComponent returns the virtual machine component for Sensor to use.
+func NewComponent() Component {
+	return &componentImpl{
+		centralReady:    concurrency.NewSignal(),
+		lock:            &sync.Mutex{},
+		stopper:         concurrency.NewStopper(),
+		virtualMachines: make(chan *storage.VirtualMachine),
```

Comment:
`virtualMachines` is currently not being closed. This is not a big issue, because we have it in other components like this, but if you would like to improve on this, you could make the channel in `Start` and close it in `Stop`.